core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -230,7 +232,16 @@ private boolean isSelectedSubnetAvailableAndRequestedType(Network network, Envir
             if (StringUtils.isNotEmpty(subnetId)) {
                 LOGGER.debug("Found selected stack subnet {}", subnetId);
                 Optional<CloudSubnet> selectedSubnet = subnetSelector.findSubnetById(envNetwork.getSubnetMetas(), subnetId);
-                if (selectedSubnet.isPresent()) {
+
+                // "noPublicIp" is an option for Azure and GCP that is used to set the network to private or public, it is not
+                // set on AWS networks. So, we check for it first, then fall back to AWS.
+                if (params.get("noPublicIp") != null) {
```

Comment:
could you use `ProviderParameterCalculator` to extract values instead of using hardcoded strings?

Reply:
also this method got big enough, should be split up

Reply:
I'd like to do this as part of [CB-12229](https://jira.cloudera.com/browse/CB-12229).

Reply:
@lacikaaa Can you help me with the `ProviderParameterCalculator`? I don't understand how to use it.

It looks like I need a `ProviderParametersBase`, which is implemented by a `*Base`, `*Request`, `*Response` class, e.g. `NetworkBase`, `NetworkRequest`, `NetworkResponse`.

The two public methods to use on `ProviderParameterCalculator` are:
* `get()`, which looks like it returns a `Mappable` representation of the provided `ProviderParameterBase source`
* `parse(Map<String, Object> parameters, ProviderParametersBase base)` which looks like it merges the `parameters` argument into the `base`?

It looks like I'd want to use `get()` and then `asMap()`, but wouldn't I still need to provide a hardcoded string as the key to my map?

Reply:
With much help from @hreeve-cloudera I reworked this to use the `ProviderParmaeterCalculator`. Needed to make a change to `NetworkV4Base` to lookup `isNoPublicIp`.

Also broke two methods out of the main `isSelectedSubnetAvailableAndRequestedType`.